File: src/Kotlin/task_3/numbers.kt
fun main() {
    println()
    val input = readLine()?.trim()?.split("\\s+".toRegex()) ?: return

    val count = input.getOrNull(0)?.toIntOrNull() ?: return
    var output = StringBuilder()

    for (i in 1 until input.size) {
        val num = input.getOrNull(i)?.toIntOrNull() ?: continue
        val (sum, product) = sumAndProductOfDigits(num)

        if (sum < product) {
            output.append("$i ")
        }
    }

    println(output.toString().trim())
}

fun sumAndProductOfDigits(number: Int): Pair<Int, Int> {
    var n = number
    var sum = 0
    var product = 1

    while (n > 0) {
        val digit = n % 10
        sum += digit
        product *= digit
        n /= 10
    }

    return Pair(sum, product)
}

File: src/JavaScript/task_3/numbers.js
function sumAndProductOfDigits(n) {
    let sum = 0, product = 1, digit;

    while (n) {
        digit = n % 10;
        sum += digit;
        product *= digit;
        n = Math.floor(n / 10);
    }

    return { sum, product };
}

function processInput(input) {
    let numbers = input.split(' ');
    let count = parseInt(numbers.shift(), 10);
    let output = ''; // Инициализируем выходную строку
    
    for (let i = 1; i <= count; i++) {
        let num = parseInt(numbers[i - 1], 10);
        let { sum, product } = sumAndProductOfDigits(num);
        
        if (sum < product) {
            output += i + ' '; // Добавляем индекс к выходной строке
        }
    }

    console.log(output.trim()); // Выводим результат и удаляем пробел в конце
}

let inputString = prompt(); 
processInput(inputString);